src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -430,28 +430,33 @@ private void RecordRequestStartMetrics(HttpContext httpContext)
             }
         }
 
+        // The trace id was successfully extracted, so we can set the trace state
+        // https://www.w3.org/TR/trace-context/#tracestate-header
         if (!string.IsNullOrEmpty(requestId))
         {
             if (!string.IsNullOrEmpty(traceState))
             {
                 activity.TraceStateString = traceState;
             }
-            var baggage = _propagator.ExtractBaggage(headers, static (object? carrier, string fieldName, out string? fieldValue, out IEnumerable<string>? fieldValues) =>
-            {
-                fieldValues = default;
-                var headers = (IHeaderDictionary)carrier!;
-                fieldValue = headers[fieldName];
-            });
+        }
 
-            // AddBaggage adds items at the beginning  of the list, so we need to add them in reverse to keep the same order as the client
-            // By contract, the propagator has already reversed the order of items so we need not reverse it again
-            // Order could be important if baggage has two items with the same key (that is allowed by the contract)
-            if (baggage is not null)
+        // Baggage can be used regardless of whether Distributed Tracing is used.
```

Comment:
```suggestion
        // Baggage can be used regardless of whether a distributed trace id was present on the inbound request.
```

Saying 'distributed tracing is used' is a little more ambiguous of a statement. Many people would probably say this code path is still using distributed tracing because it created an Activity object. I think what you wanted to distinguish is that sometimes the inbound request provides us the id to use and sometimes we create our ID on-demand. Baggage can work in either case.